daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,116 @@
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+if [ ! -z "$(which systemd-detect-virt 2>/dev/null || command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+else
+    # /proc/1/sched exposes the host's pid of our init !
+    # http://stackoverflow.com/a/37016302
+    pid=$( cat /proc/1/sched 2>/dev/null | head -n 1 | { IFS='(),#:' read name pid th threads; echo $pid; } )
+    if [ ! -z "${pid}" ]; then
+        pid=$(( pid + 0 ))
+        [ ${pid} -ne 1 ] && VIRTUALIZATION="container"
+    fi
+
+    # ubuntu and debian supply /bin/running-in-container
+    # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+    [ -x "/bin/running-in-container" ] && "/bin/running-in-container" >/dev/null 2>&1 && VIRTUALIZATION="container"
+
+    # lxc sets environment variable 'container'
+    [ ! -z "${container}" ] && VIRTUALIZATION="lxc"
+
+    # docker creates /.dockerenv
+    # http://stackoverflow.com/a/25518345
+    [ -f "/.dockerenv" ] && VIRTUALIZATION="docker"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata version
+
+if [ -z "${NETDATA_VERSION}" ]; then
+    NETDATA_VERSION="$(netdata -V 2>&1 | cut -d ' ' -f 2)"
+fi
+
+if [ -z "${NETDATA_REGISTRY_UNIQUE_ID}" ]; then
+    for dir in "$(netdata -W get global 'lib directory' '/var/lib/netdata' 2>/dev/null)" "/var/lib/netdata" "/opt/netdata/var/lib/netdata"; do
+        if [ -f "${dir}/registry/netdata.public.unique.id" ]; then
+            NETDATA_REGISTRY_UNIQUE_ID="$(cat "${dir}/registry/netdata.public.unique.id")"
+            break
+        fi
+    done
+fi
+
+OPTOUT=
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/etc/netdata" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            OPTOUT="$(cat "${dir}/.opt-out-from-anonymous-statistics")"
+            break
+        fi
+    done
+else
+    OPTOUT="$(cat "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics")"
+fi
+
+if [ "${OPTOUT}" = "all" ] || [ "${OPTOUT}" = "usage" -a "${ACTION}" = "START" ]; then
+    # the user does not want us to send statistics
+    exit 0
+fi
+
+# -------------------------------------------------------------------------------------------------
+# send the anonymous statistics to netdata
+
+# curl -Ss >/dev/null 2>&1 --max-time 5
+echo "https://registry.my-netdata.io/log/anonymous-statistics?v=1&version=${NETDATA_VERSION}&machine_guid=${NETDATA_REGISTRY_UNIQUE_ID}&os_detection=${OS_DETECTION}&distro_name=${NAME}&distro_id=${ID}&distro_id_like=${ID_LIKE}&distro_version=${VERSION}&distro_version_id=${VERSION_ID}&kernel_name=${KERNEL_NAME}&kernel_version=${KERNEL_VERSION}&architecture=${ARCHITECTURE}&virtualization=${VIRTUALIZATION}"
```

Comment:
Telemetry data shouldn't be passed as GET, it should be POST.

Reply:
yes, I know. It is just a test to see it currently.
There is nothing listening there to collect them...

Reply:
added it, commented though